app/scripts/services/rest.js

```diff
@@ -57,4 +52,35 @@ angular.module('openhimWebui2App')
       }),
 
     };
+  })
+  .factory('Import', function ($rootScope, $resource) {
+
+    // fetch API server details
+    var protocol = angular.copy( $rootScope.protocol );
+    var host = angular.copy( $rootScope.host );
+    var port = angular.copy( $rootScope.port );
+    var server = protocol + '://' + host + ':' + port;
+
+    return {
+      Channels: $resource( server + '/channels/:channelId', { channelId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
+
+      Users: $resource( server + '/users/:userId', { userId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
+
+      Clients: $resource( server + '/clients/:clientId/', { clientId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
+
+      Mediators: $resource( server + '/mediators:mediatorId/', { mediatorId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
+
+      ContactGroups: $resource( server + '/groups/:groupId', { groupId: '@_id' }, {
+        update: { method: 'PUT' }
+      }),
```

Comment:
I had some issues with the API not being consistent with the imports. Namely Users and Mediators, They do not use '_id' as their reference key so the imports would always fail because for Users e.g email would always be attached to the endpoint which is not what needs to happen for the import. 

I created the separate import factory as the current endpoints were not all compatable with the import process and to separate the diff kind ofendpoints and it allows for more control over the import